tests/components/synology_dsm/test_config_flow.py

```diff
@@ -289,6 +278,36 @@ async def test_login_failed(hass: HomeAssistantType, service_login_failed: Magic
     assert result["errors"] == {CONF_USERNAME: "login"}
 
 
+async def test_connection_failed(hass: HomeAssistantType, service: MagicMock):
+    """Test when we have errors during connection."""
+    service.return_value.login = Mock(
+        side_effect=SynologyDSMRequestException(IOError("arg"))
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_HOST: "connection"}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5403678)